kotishell/src/tokenize/tokenize_d.c

```diff
@@ -74,6 +74,6 @@ t_tk_symbol_type	l_tk_get_symbol_type(char c)
 
 void	l_tk_symbol_error(t_tk_so_far *so_far)
 {
-	mini_putstr_fd((const char *)so_far, 2);
+	write(1, so_far->token.data, so_far->token.length + 4);
```

Comment:
Two things:

1. Should we write error messages to STDERR (2) instead of STDOUT (1)?
2. writing 4 chars might be too much in some cases. We might access the memory we don't own if the unsupported symbol is shorter than 3 bytes and is located in the end. For example, a unicode cyrillic letter is only 2 bytes.

I see why writing 4 bytes is nice. Outputting only the first character on an emoji is gibberish. But avoiding out of bounds access is important, too. And the length of 4 doesn't fit all the weird stuff. Let's do min(4, number of remaining characters) to avoid out of memory access and still have a nice result for emojis and something reasonable for cyrillic letters. We can add a switch statement that tries to detect the length of a unicode symbol later.

Reply:
agreed and updated